File: backend/app/models/order.py
from typing import List, Optional
from pydantic import BaseModel, Field
from datetime import datetime

class OrderItemInput(BaseModel):
    product_id: str
    quantity: int = Field(..., gt=0, description="Order quantity (must be > 0)")

class OrderCalculateRequest(BaseModel):
    store_slug: str
    items: List[OrderItemInput]
    customer_name: Optional[str] = Field(None, description="Customer name for personalized WhatsApp order")
    customer_phone: Optional[str] = Field(None, description="Customer contact phone")
    delivery_notes: Optional[str] = Field(None, description="Delivery location / instructions")

class CalculatedOrderItem(BaseModel):
    product_id: str
    title: str
    unit_price: float
    quantity: int
    subtotal: float
    available_stock: int
    is_sufficient: bool

class OrderCalculateResponse(BaseModel):
    store_name: str
    whatsapp_number: str
    items: List[CalculatedOrderItem]
    total_amount: float
    currency: str = "LKR"
    has_stock_issues: bool
    stock_warnings: List[str] = []
    whatsapp_checkout_url: str
    formatted_whatsapp_message: str

class OrderCreateRequest(OrderCalculateRequest):
    save_to_store: bool = True

class OrderRecordResponse(BaseModel):
    id: str
    store_id: str
    customer_name: Optional[str] = None
    customer_phone: Optional[str] = None
    total_amount: float
    status: str
    delivery_notes: Optional[str] = None
    created_at: Optional[datetime] = None

File: Rust/src/models/order.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct OrderItemInput {
    pub product_id: String,
    pub quantity: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct OrderCalculateRequest {
    pub store_slug: String,
    pub items: Vec<OrderItemInput>,
    #[serde(default)]
    pub customer_name: Option<String>,
    #[serde(default)]
    pub customer_phone: Option<String>,
    #[serde(default)]
    pub delivery_notes: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CalculatedOrderItem {
    pub product_id: String,
    pub title: String,
    pub unit_price: f64,
    pub quantity: i64,
    pub subtotal: f64,
    pub available_stock: i64,
    pub is_sufficient: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct OrderCalculateResponse {
    pub store_name: String,
    pub whatsapp_number: String,
    pub items: Vec<CalculatedOrderItem>,
    pub total_amount: f64,
    pub currency: String,
    pub has_stock_issues: bool,
    pub stock_warnings: Vec<String>,
    pub whatsapp_checkout_url: String,
    pub formatted_whatsapp_message: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct OrderCreateRequest {
    pub store_slug: String,
    pub items: Vec<OrderItemInput>,
    #[serde(default)]
    pub customer_name: Option<String>,
    #[serde(default)]
    pub customer_phone: Option<String>,
    #[serde(default)]
    pub delivery_notes: Option<String>,
    #[serde(default = "default_true")]
    pub save_to_store: bool,
}

fn default_true() -> bool {
    true
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct OrderRecordResponse {
    pub id: String,
    pub store_id: String,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub customer_name: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub customer_phone: Option<String>,
    pub total_amount: f64,
    pub status: String,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub delivery_notes: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub created_at: Option<String>,
}
